scripts/server_with_rewrites.py

```diff
@@ -1,16 +1,28 @@
 from http.server import HTTPServer, SimpleHTTPRequestHandler
 import argparse
 
+class MyHTTPServer(HTTPServer):
+    def __init__(self, server_address, RequestHandlerClass,
+                 base_path='',
+                 bind_and_activate=True):
+        HTTPServer.__init__(self, server_address, RequestHandlerClass, bind_and_activate=bind_and_activate)
+        self.base_path = base_path
 
-class RequestHandler(SimpleHTTPRequestHandler):
 
+class RequestHandler(SimpleHTTPRequestHandler):
     # mimic firebase hosting's rewrite rules
-    def translate_path(self, path):
-        if path.startswith('/simplified') or path.startswith('/traditional') or path.startswith('/cantonese') or path.startswith('/hsk'):
+    def translate_path(self, request_path):
+        base_path = self.server.base_path if isinstance(self.server, MyHTTPServer) else ''
+        slashed_path = request_path.removeprefix(base_path)
+        crit_blank = slashed_path in ('/', '')
+        crit_specials = any([slashed_path.startswith(f"{candidate}") for candidate in (
+            '/simplified', '/traditional', '/cantonese', '/hsk'
+        )])
+        if crit_blank or crit_specials:
             return 'index.html'
-        # prepend . such that the frontend's use of /whatever results in ./whatever
-        # should maybe just strip the first character, tbh
-        return f".{path}"
+        # now turn the slashed path /foo/bar into ./foo/bar on disk
+        path = f".{slashed_path}"
```

Comment:
I realized I should ask: how widely exposed would this be? I tried a little testing of path traversal attacks, and it seems robust to them, but I'd just be careful if you're planning on exposing it to everyone on the internet.

Granted, your docker setup would help with that as well. Anyway, just figured I'd mention it in case.